src/librustc/mir/mod.rs

```diff
@@ -395,6 +411,10 @@ pub struct LocalDecl<'tcx> {
     /// True if this corresponds to a user-declared local variable.
     pub is_user_variable: bool,
 
+    /// True if this an internal local.
+    /// Such locals are not checked against the legal types in a generator.
+    pub internal: bool,
```

Comment:
At least have it a well-documented flag then. MIR is "semi-public", and its semantics are supposed to be fairly clear.